Solutions/Menes.Hosting.AspNetCore/Menes.Hosting.AspNetCore.csproj

```diff
@@ -9,16 +9,23 @@
     <PackageDescription>Menes is a framework for hosting Web APIs with OpenAPI-based service definitions. This library defines the ASP.NET-Core-specific aspects of hosting in Menes.</PackageDescription>
 
     <WarningsAsErrors />
+
+    <!-- Don't want ConfigureAwait warnings because ASP.NET Core doesn't use synchronization context, so such calls are overhead with no purpose. -->
+    <NoWarn>$(NoWarn);RCS1090</NoWarn>
   </PropertyGroup>
 
+  <ItemGroup>
+    <InternalsVisibleTo Include="Menes.Specs" />
+  </ItemGroup>
+
   <ItemGroup>
     <PackageReference Include="Endjin.RecommendedPractices" Version="1.2.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
```

Comment:
Should we not be switching this to use the aspnet project type rather than referencing these aspnetcore DLLs?

Reply:
I don't know if that would cause issues for Azure Functions, because that uses ASP.NET Core, but isn't an ASP.NET Core app.

Reply:
On further investigation, it turns out we need to drop support for ASP.NET Core 2.1. I'm not convinced we really supported ASP.NET Core 2.1 anyway, what with the relevant `<PackageReference>`s all being to v2.2.x, a version that has technically been out of support for some time.

So I've modified the relevant projects to target `netcoreapp3.1`, which has enabled these hokey direct package references to be replaced with a single `<FrameworkReference>` to `Microsoft.AspNetCore.App` which is how you're now supposed to do this in a library that needs to reference such components, but which doesn't need any of the ASP.NET Core build mecahnisms.